### DISCUSSION TITLE
How to run CLIENT URL and API URL at the same time in React

### DISCUSSION THREAD

**Discussion by sharonanderson-dev** [community/community#36501] [Actions] [answered]
So I want client and server simultaneously on my React app. But I can run only the client URL and not the server is there any way to solve this issue. 

I used to run my code using npm
```
npm start
```

My code looks something like this.
![Screenshot 2022-10-18 072345](https://user-images.githubusercontent.com/92415144/196317610-e614ba51-cd6f-47e5-8e5d-e0a4deeadd97.png)

**Comment by santosharron** [comment, marked as answer]
It's pretty easier. Let's say you have an Client URL as `http://localhost:3000/` and Server URL as `http://localhost:4242/`. I like [task automation with npm](https://github.com/substack/blog/blob/master/npm_run.markdown) but the usual way to run multiple commands concurrently is `npm run watch-js & npm run watch-css`. That's fine but it's hard to keep on track of different outputs. Also if one process fails, others keep running and you won't even notice the difference.
 Another option would be just to run all commands in separate terminals. I got tired of opening terminals and making them concurrently.

The tool is written in Node.js, but you can use it to run any commands.
```
npm i concurrently
```
or 
```
npm install concurrently
```
Let's say you have a `package.json` script something like this...
```json 
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
```

Now let's change that to something like this...

```json
  "homepage": "http://localhost:3000/",
  "proxy": "http://localhost:4242",
  "scripts": {
    "start-client": "react-scripts start",
    "start-server": "node server.js",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "start": "concurrently \"yarn start-client\" \"yarn start-server\""
  },
```
And That's it Now it should work 😊

**Reply by sharonanderson-dev** [reply]
Yea, it worked Thanks for the help!
